test/spec/vis/vis.spec.js

```diff
@@ -974,6 +975,36 @@ describe('vis/vis', function () {
       this.vis = new Vis();
     });
 
+    describe('remove dataview', function () {
+      beforeEach(function () {
+        spyOn(this.vis, 'reload');
+        this.vis.load(new VizJSON(fakeVizJSON()));
+        this.vis.instantiateMap();
+      });
+
+      it('should reload vis', function () {
```

Comment:
Also, we could test that the vis is not reloaded when the dataview is not filtered.